### ISSUE TITLE
New API for ML.NET

### ISSUE THREAD

**Issue by Zruty0** [dotnet/machinelearning#754]
We are creating a stable API that:
- Uses parallel terminology with other well-known ML libraries (Spark, sklearn);
- Takes advantage of strong types of .NET to shorten path to success;
- Is going to be present from now till 1.0 and beyond;
- Keeps simple ML scenarios concise;
- Allows advanced ML scenarios: see #584 .

To that end, we are going to expose a selection of Estimators and Transformers (see #581 ) that cover existing transforms, learners and loaders. 

This issue will be used to track the overall [project](https://github.com/dotnet/machinelearning/projects/9) status: what is planned to be done, what is done, etc.

**Comment by Zruty0** [comment]
List of trainers:

| Trainer | Category | Priority | Status |
| ------- | -------- | -------- | ------ |
| KMeansPlusPlus: KMeans++ Clustering | Clustering | 0 | |
| LogisticRegression: Logistic Regression | Linear | 0 | |
| MultiClassLogisticRegression: Multi-class Logistic Regression | Linear | 0 | |
| OLSLinearRegression: Ordinary Least Squares (Regression) | Linear | 0 | |
| SDCA: Fast Linear (SA-SDCA) | Linear | 0 | |
| SDCAMC: Fast Linear Multi-class Classification (SA-SDCA) | Linear | 0 | |
| SDCAR: Fast Linear Regression (SA-SDCA) | Linear | 0 | |
| SymbolicSGD: Symbolic SGD (binary) | Linear | 0 | |
| FastForestClassification: Fast Forest Classification | Tree | 0 | |
| FastForestRegression: Fast Forest Regression | Tree | 0 | |
| FastTreeBinaryClassification: FastTree (Boosted Trees) Classification | Tree | 0 | |
| FastTreeRegression: FastTree (Boosted Trees) Regression | Tree | 0 | |
| LightGBMBinary: LightGBM Binary Classifier | Tree | 0 | |
| LightGBMMulticlass: LightGBM Multi-class Classifier | Tree | 0 | |
| LightGBMRegression: LightGBM Regressor | Tree | 0 | |
| FieldAwareFactorizationMachine: Field-aware Factorization Machine | FFM | 1 | |
| BinaryClassificationGamTrainer: Generalized Additive Model for Binary Classification | GAM | 1 | |
| RegressionGamTrainer: Generalized Additive Model for Regression | GAM | 1 | |
| AveragedPerceptron: Averaged Perceptron | Linear | 1 | |
| OnlineGradientDescent: Stochastic Gradient Descent (Regression) | Linear | 1 | |
| oissonRegression: Poisson Regression | Linear | 1 | |
| OVA: One-vs-All | Meta | 1 | |
| PKPD: Pairwise coupling (PKPD) | Meta | 1 | |
| pcaAnomaly: PCA Anomaly Detector | Projection | 1 | |
| FastTreeTweedieRegression: FastTree (Boosted Trees) Tweedie Regression | Tree | 1 | |
| PriorPredictor: Prior Predictor | Baseline | 2 | |
| RandomPredictor: Random Predictor | Baseline | 2 | |
| MultiClassNaiveBayes: Multiclass Naive Bayes | Bayes | 2 | |
| BinarySGD: Hogwild SGD (binary) | Linear | 2 | |
| LinearSVM: SVM (Pegasos-Linear) | Linear | 2 | |
| EnsembleRegression: Regression Ensemble (bagging, stacking, etc) | Meta | 2 | |
| WeightedEnsemble: Parallel Ensemble (bagging, stacking, etc) | Meta | 2 | |
| WeightedEnsembleMulticlass: Multi-class Parallel Ensemble (bagging, stacking, etc) | Meta | 2 | |
| FastTreeRanking: FastTree (Boosted Trees) Ranking | Tree | 2 | |
| LightGBMRanking: LightGBM Ranking | Tree | 2 | |

**Comment by shauheen** [comment]
@Ivanidzo4ka can you please verify if this can be closed now?

**Comment by eerhardt** [comment]
I believe this can be closed. We have made this new API and have separate issues tracking remaining necessary API work.

Please re-open if I'm wrong.
